gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +734,689 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+
+void FITSDataset::WriteFITSInfo()
+
+{
+  int status = 0;
+
+  const double PI = std::atan(1.0)*4;
+  const double DEG2RAD = PI / 180.;
+
+  double falseEast = 0;
+  double falseNorth = 0;
+
+  double cfactor, mres, mapres, UpperLeftCornerX, UpperLeftCornerY;
+  double crpix1, crpix2;
+
+/* -------------------------------------------------------------------- */
+/*      Write out projection definition.                                */
+/* -------------------------------------------------------------------- */
+    const bool bHasProjection = !oSRS.IsEmpty();
+    if( bHasProjection )
+    {
+
+        // Set according to coordinate system (thanks to Trent Hare - USGS)
+
+        std::string object, ctype1, ctype2;
+        
+        const char* target = oSRS.GetAttrValue("DATUM",0);
+        if ( target ) {
+            if ( strstr(target, "Moon") ) {
+              object.assign("Moon");
+              ctype1.assign("SE");
+              ctype2.assign("SE");              
+            } else if ( strstr(target, "Mercury") ) {
+              object.assign("Mercury");
+              ctype1.assign("ME");
+              ctype2.assign("ME");
+            } else if ( strstr(target, "Venus") ) {
+              object.assign("Venus");
+              ctype1.assign("VE");
+              ctype2.assign("VE");
+            } else if ( strstr(target, "Mars") ) {
+              object.assign("Mars");
+              ctype1.assign("MA");
+              ctype2.assign("MA");
+            } else if ( strstr(target, "Jupiter") ) {
+              object.assign("Jupiter");
+              ctype1.assign("JU");
+              ctype2.assign("JU");
+            } else if ( strstr(target, "Saturn") ) {
+              object.assign("Saturn");
+              ctype1.assign("SA");
+              ctype2.assign("SA");
+            } else if ( strstr(target, "Uranus") ) {
+              object.assign("Uranus");
+              ctype1.assign("UR");
+              ctype2.assign("UR");
+            } else if ( strstr(target, "Neptune") ) {
+              object.assign("Neptune");
+              ctype1.assign("NE");
+              ctype2.assign("NE");
+            } else {
+              object.assign("Earth");
+              ctype1.assign("EA");
+              ctype2.assign("EA");
+            }
+
+            char * cstrobj = new char [object.length()+1];
+            std::strcpy (cstrobj, object.c_str());
+
+            fits_update_key( hFITS, TSTRING, "OBJECT", cstrobj, nullptr, &status);
+        }
+
+        double aradius = oSRS.GetSemiMajor();
+        double bradius = aradius;
+        double cradius = oSRS.GetSemiMinor();
+
+        cfactor = aradius * DEG2RAD;
+
+        fits_update_key( hFITS, TDOUBLE, "A_RADIUS", &aradius, nullptr, &status);
+        fits_update_key( hFITS, TDOUBLE, "B_RADIUS", &bradius, nullptr, &status);
+        fits_update_key( hFITS, TDOUBLE, "C_RADIUS", &cradius, nullptr, &status);
+
+        const char* un = oSRS.GetAttrValue("UNIT",0);
```

Comment:
please unit `unit` which is much more descriptive

Reply:
Fixed in https://github.com/OSGeo/gdal/pull/1298/commits/f17edf20f02f51decbbc735b0dfe0c4219a20111